File: roman to integer/roman_to_integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        sum = 0;
        sum+=len(s.split('I'))-1
        sum+=5*(len(s.split('V'))-1)
        sum+=10*(len(s.split('X'))-1)
        sum+=50*(len(s.split('L'))-1)
        sum+=100*(len(s.split('C'))-1)
        sum+=500*(len(s.split('D'))-1)
        sum+=1000*(len(s.split('M'))-1)
        sum-=2*(len(s.split('IV'))-1)
        sum-=2*(len(s.split('IX'))-1)
        sum-=20*(len(s.split('XL'))-1)
        sum-=20*(len(s.split('XC'))-1)
        sum-=200*(len(s.split('CD'))-1)
        sum-=200*(len(s.split('CM'))-1)
        return sum
        
        
            

File: roman to integer/RomanToInteger.java
class RomanToInteger {
    public int romanToInt(String s) {
        int sum = 0;
        char [] charr = s.toCharArray();
        for(int i = 0; i< s.length();i++){
            switch(charr[i]){
                case 'I':
                    
                    if(i != s.length()-1 && (charr[i+1]=='V' || charr[i+1]=='X')) sum-=1;
                    else sum+=1;
                    break;
                case 'V':
                    sum+=5;
                    break;
                case 'X':
                    if(i!= s.length()-1 && (charr[i+1]=='L' || charr[i+1]=='C')) sum -=10;
                    else sum+=10;
                    break;
                case 'L':
                    sum+=50;
                    break;
                case 'C':
                    if(i!= s.length()-1 && (charr[i+1]=='D' || charr[i+1]=='M')) sum -=100;
                    else sum+=100;
                    break;
                case 'D':
                    sum+=500;
                    break;
                case 'M':
                    sum+=1000;
                    break;
            }
        }
        return sum;
    }
}